runtime/src/commonMain/kotlin/pbandk/internal/json/JsonMessageEncoder.kt

```diff
@@ -38,8 +38,9 @@ internal class JsonMessageEncoder(private val jsonConfig: JsonConfig) : MessageE
             @Suppress("UNCHECKED_CAST")
             val value = (fd.value as KProperty1<T, *>).get(message)
 
-            if (value == null && fd.oneofMember) continue
-            if (!fd.oneofMember && !jsonConfig.outputDefaultValues && fd.type.isDefaultValue(value)) continue
+            val oneOfOrExplicitPresence = (fd.oneofMember || fd.type.hasPresence)
```

Comment:
If I remember right, I think `fd.type.hasPresence` will always be `true` for oneof fields. So you might be able to simplify this and only test for `fd.type.hasPresence` and drop the explicit test for `fd.oneofMember`.

Reply:
I applied the change, thank you. The tests that we have still pass.

I tried quickly to trace where `hasPresence` is being set for oneOf fields but I couldn't find it easily :) I would appreciate a pointer so I can better understand what I'm doing :)

Reply:
`hasPresence` in the generated code is set here: https://github.com/streem/pbandk/blob/1fa4fdac0396f6ddfd327265d31c322348baa38a/protoc-gen-pbandk/lib/src/commonMain/kotlin/pbandk/gen/CodeGenerator.kt#L536. That is in turn based on the value of `File.Field.Numbered.hasPresence`, which is set here during the parsing of the proto files: https://github.com/streem/pbandk/blob/1fa4fdac0396f6ddfd327265d31c322348baa38a/protoc-gen-pbandk/lib/src/commonMain/kotlin/pbandk/gen/FileBuilder.kt#L190. That's where the oneof-related logic is coming from.